src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithPlainTextCommentFilterTest.java

```diff
@@ -390,13 +455,93 @@ public void testSuppressByModuleId() throws Exception {
+                ".*[a-zA-Z][0-9].*"),
+            "11: " + getCheckMessage(RegexpSinglelineCheck.class, MSG_REGEXP_EXCEEDED,
+                ".*[a-zA-Z][0-9].*"),
+            "14: " + getCheckMessage(RegexpSinglelineCheck.class, MSG_REGEXP_EXCEEDED,
+                ".*[a-zA-Z][0-9].*"),
+        };
+
+        verifySuppressed(
+            "InputSuppressWithPlainTextCommentFilterSuppressById.java",
+            removeSuppressed(expectedViolationMessages, suppressedViolationMessages),
+            filterCfg, regexpCheckCfg, fileTabCheckCfg
+        );
+    }
+
+    @Test
+    public void testSuppressByCheckAndNonMatchingModuleId() throws Exception {
+        final DefaultConfiguration filterCfg =
+            createModuleConfig(SuppressWithPlainTextCommentFilter.class);
+        filterCfg.addAttribute("offCommentFormat", "CSOFF (\\w+) \\(\\w+\\)");
+        filterCfg.addAttribute("onCommentFormat", "CSON (\\w+)");
+        filterCfg.addAttribute("checkFormat", "FileTabCharacterCheck");
+        filterCfg.addAttribute("idFormat", "$1");
+
+        final DefaultConfiguration regexpCheckCfg = createModuleConfig(RegexpSinglelineCheck.class);
+        regexpCheckCfg.addAttribute("id", "ignore");
+        regexpCheckCfg.addAttribute("format", ".*[a-zA-Z][0-9].*");
+
+        final DefaultConfiguration fileTabCheckCfg =
+            createModuleConfig(FileTabCharacterCheck.class);
+        fileTabCheckCfg.addAttribute("eachLine", "true");
+        fileTabCheckCfg.addAttribute("id", "foo");
+
+        final String[] suppressedViolationMessages = CommonUtil.EMPTY_STRING_ARRAY;
+
+        final String[] expectedViolationMessages = {
+            "6: " + getCheckMessage(RegexpSinglelineCheck.class, MSG_REGEXP_EXCEEDED,
+                ".*[a-zA-Z][0-9].*"),
+            "9:1: " + getCheckMessage(FileTabCharacterCheck.class, MSG_CONTAINS_TAB),
+            "9: " + getCheckMessage(RegexpSinglelineCheck.class, MSG_REGEXP_EXCEEDED,
+                ".*[a-zA-Z][0-9].*"),
+            "11: " + getCheckMessage(RegexpSinglelineCheck.class, MSG_REGEXP_EXCEEDED,
+                ".*[a-zA-Z][0-9].*"),
+            "14: " + getCheckMessage(RegexpSinglelineCheck.class, MSG_REGEXP_EXCEEDED,
+                ".*[a-zA-Z][0-9].*"),
+        };
+
+        verifySuppressed(
+            "InputSuppressWithPlainTextCommentFilterSuppressById.java",
+            removeSuppressed(expectedViolationMessages, suppressedViolationMessages),
+            filterCfg, regexpCheckCfg, fileTabCheckCfg
+        );
+    }
+
     @Test
     public void testSuppressByModuleIdWithNullModuleId() throws Exception {
         final DefaultConfiguration filterCfg =
             createModuleConfig(SuppressWithPlainTextCommentFilter.class);
         filterCfg.addAttribute("offCommentFormat", "CSOFF (\\w+) \\(\\w+\\)");
         filterCfg.addAttribute("onCommentFormat", "CSON (\\w+)");
-        filterCfg.addAttribute("checkFormat", "$1");
+        filterCfg.addAttribute("idFormat", "$1");
```

Comment:
changed `checkFormat` to `idFormat` because test is `testSuppressByModuleIdWithNullModuleId` and was using `checkFormat` to match by ID